### PR TITLE
Alan fix total summaries

### PR DESCRIPTION
**Pull request by LeeSingChanYau**
Fix for Weekly summaries view for volunteers is missing last week (see below), so we’ll have 4 tabs instead of 3 from now on. Also, the “total submitted” is not accurate. Please add another tab showing the end date for last week.

### CODE REVIEW

**Comment by rauleffting** [comment]
@LeeSingChanYau I ran your PR locally, but the 'last week' tab is still missing.  It's missing not just for volunteers, but for all account types as well.

Here is a screenshot:
Screen Shot 2023-01-02 at 15 36 08

**Comment by LeeSingChanYau** [comment]
Hey Raul, thanks for catching that. I made a change please review again and tell if it looks good.

**Comment by rauleffting** [comment]
Hi @LeeSingChanYau ! Thank you very much for taking a look. I've tested again and I found a new error. Now, it's possible to see the tabs, but when I add a new tangible or intangible time, it doesn't appear on 'Current Week Timelog' .

Here is a video:

Uploading Screen Recording 2023-01-07 at 12.03.22.mov…

**Comment by rauleffting** [comment]
Hi @LeeSingChanYau ! Thank you very much for taking a look. I've tested again and I found a new error. Now, it's possible to see the tabs, but when I add a new tangible or intangible time, it doesn't appear on 'Current Week Timelog' .

Here is a video:

https://user-images.githubusercontent.com/29555732/211157843-c48bb2eb-5b1b-430d-b015-5682ace391af.mov

**Comment by LeeSingChanYau** [comment]
Hi @rauleffting, yes you are right that is not behaving correctly. But I think that is the Timelog component. In this PR I am working on the WeeklySummary Component

**Comment by LeeSingChanYau** [comment]
image

Maybe you can add this to the bugs google doc.

**Comment by LeeSingChanYau** [comment]
> image
> 
> Maybe you can add this to the bugs google doc.

As you can see above they are two different components.

**Comment by one-community** [comment]
> 

It should have 4 tabs, not 3 too. So still missing "2 Weeks Ago" tab.

**Comment by LeeSingChanYau** [comment]
ok I can do that, but I'm not sure how to access to the summary before 2 weeks ago.

**Comment by one-community** [comment]
Are they not stored in the database? Maybe see if Yiyun can help you.

**Comment by one-community** [comment]
Following up on this

**Comment by rauleffting** [comment]
> > image
> > Maybe you can add this to the bugs google doc.
> 
> As you can see above they are two different components.

**Comment by rauleffting** [comment]
@LeeSingChanYau You are right and I was wrong about the component. I'll take a look at your new changes. 

p.s: Sorry, I closed it by mistake.

**Comment by rauleffting** [comment]
Hi, @LeeSingChanYau! I tested your PR. As you can see in the video below, the weekly summaries are showing as expected. However, there are two things that must be fixed:
1. the fourth tab is showing the wrong date because '3 weeks ago' wasn't at first January. I also think that the tab title is supposed to be 'Three weeks ago'. Please, correct me if I'm wrong.
2. the second thing is that when a new weekly summary is added, the page freezes and it's necessary to reload.

https://user-images.githubusercontent.com/29555732/212479909-ce137d7a-1533-46d6-b5ca-e38614f50d23.mov

Here are the errors that the console shows when clicking on save:

Screen Shot 2023-01-14 at 12 33 51

Thank you!

**Comment by LeeSingChanYau** [comment]
Hello @rauleffting, yes as I said in slack the weeklySummaries are not loading correctly since the first of january in the dev branch. Things are working fine in the main branch, but in the dev the 3 summaries are displaying the due dates for this week.
Screenshot 2023-01-14 at 8 40 30 PM

**Comment by LeeSingChanYau** [comment]
Also as of right now we cant really show the last 4 summaries because the database only stores the last 3, so unless we merge a new pr that stores 4 summaries this will still be happening.

**Comment by one-community** [comment]
Please create that PR for a 4th summary

**Comment by rauleffting** [comment]
> Hello @rauleffting, yes as I said in slack the weeklySummaries are not loading correctly since the first of january in the dev branch. Things are working fine in the main branch, but in the dev the 3 summaries are displaying the due dates for this week. Screenshot 2023-01-14 at 8 40 30 PM

Ok. But we need to be sure that this error would not affect the main branch if PR is merged. 

Please, change the variables' names to the same name used in the UX. So the code can be more readable. 

Thank you very much!

**Comment by LeeSingChanYau** [comment]
Hi @rauleffting, please review when you can. I made some changes.

**Comment by rauleffting** [comment]
> Hi @rauleffting, please review when you can. I made some changes.

Hi @LeeSingChanYau. I tested your PR again and it's still freezing when a summary is submitted:

https://user-images.githubusercontent.com/29555732/213874902-dbb6b43b-2a6e-4ce7-877d-dbbc01b121cf.mov

Please, change the variables summaryBeforeLast for summaryTwoWeeksAgo, like the UX:

Screen Shot 2023-01-21 at 12 47 33

Thank you very much!

**Comment by RafaelCastilhoCastro** [comment]
Hi @LeeSingChanYau,
I also tested the PR and this is the result.
image

The screen is algo freezing after submitting a new weekly summary.

**Comment by LeeSingChanYau** [comment]
Hello @rauleffting, yes you are correct nothing refreshes when we save a summary that is not for the current. But if you submit a summary for the current week, it should refresh like the video I shared below.

**Comment by LeeSingChanYau** [comment]
Thanks for catching that. That was happening because somehow I did not pass the setSubmittedSummary as prop. Please review again. Thanks again!

**Comment by PrabhjotSembhi** [comment]
Hey, I have checked it. When We add a summary we can add the last 3 weeks. But I am not sure about the date. 
If the third date shows the week's ending date the week it's ok. 

image

and in the Admin role week, summary reports it's not showing the 3rd-week reports. 

image

maybe you can do this later.

**Review by PrabhjotSembhi — CHANGES_REQUESTED** [review]
Hey, I have checked it. When We add a summary we can add the last 3 weeks. But I am not sure about the date. 
If the third date shows the week's ending date the week it's ok. 

image

and in the Admin role week, summary reports it's not showing 3rd-week reports. 

image

**Comment by rauleffting** [comment]
Hi @LeeSingChanYau, I tested your PR again. Now, the page isn't freezing anymore. However, the Three Weeks Ago tab is still not displaying the right date. I also think that it should be called Three Weeks Ago, following the pattern of the other tabs.

Screen Shot 2023-01-24 at 13 14 48

Thank you very much! Thanks to change the variable's name!

**Comment by RafaelCastilhoCastro** [comment]
Hi @LeeSingChanYau,
I performed new tests and all features seems to be working as intended. Tks.

**Review by PrabhjotSembhi — COMMENTED** [review]
In the Admin view, we are not using dates to represent weeks. 
In the volunteer tab, we are using the date in the last tab. we make it consistent, that will be better and easy to understand

**Review by PrabhjotSembhi — APPROVED** [review]

**Comment by rauleffting** [comment]
Hi @PrabhjotSembhi. Why did you approve it? It's displaying the wrong date in the last tab. Thanks.

**Comment by rauleffting** [comment]
The date remains incorrect. Cannot be approved. Thank you all.

**Comment by LeeSingChanYau** [comment]
@rauleffting please review now, I made some changes.

**Comment by LeeSingChanYau** [comment]
This is what I am getting now after saving the last three summaries.
Screenshot 2023-02-02 at 12 29 43 PM
Screenshot 2023-02-02 at 12 29 04 PM
Screenshot 2023-02-02 at 12 28 55 PM

**Review by rauleffting — APPROVED** [review]
Hi @LeeSingChanYau! I tested your changes. Everything worked quite well, as you can see in the video below:

https://user-images.githubusercontent.com/29555732/216379228-a6bdce09-6456-467e-adea-b68346b602f7.mp4

File: src/components/WeeklySummary/WeeklySummary.jsx [hunk 1]
```diff
@@ -70,33 +77,40 @@ export class WeeklySummary extends Component {
...
+      (weeklySummaries && weeklySummaries[3] && weeklySummaries[3].summary) || '';
 
     const dueDateThisWeek = weeklySummaries && weeklySummaries[0] && weeklySummaries[0].dueDate;
     // Make sure server dueDate is not before the localtime dueDate.
     const dueDate = moment(dueDateThisWeek).isBefore(this.state.dueDate)
       ? this.state.dueDate
       : dueDateThisWeek;
     const dueDateLastWeek =
-      (weeklySummaries && weeklySummaries[1] && weeklySummaries[1].dueDate) ||
+      // (weeklySummaries && weeklySummaries[1] && weeklySummaries[1].dueDate) ||
```
Comment (rauleffting) [inline src/components/WeeklySummary/WeeklySummary.jsx:91]: Please, delete these comments.

**Comment by rauleffting** [comment]
Hi @LeeSingChanYau. Thank you very much. As you can see in the video below, your changes worked quite well:

https://user-images.githubusercontent.com/29555732/216380957-482aac0f-20a0-4e7e-8dfd-a325c8bdb6f6.mp4

File: src/components/WeeklySummary/WeeklySummary.jsx (hunk 1 above)
Comment (rauleffting) [inline src/components/WeeklySummary/WeeklySummary.jsx:91]: Please, @LeeSingChanYau, delete these comments. Thereafter, I'm going to approve your PR! Thank you!

File: src/components/WeeklySummary/WeeklySummary.jsx [hunk 2]
```diff
@@ -70,33 +77,40 @@ export class WeeklySummary extends Component {
...
       this.state.dueDateLastWeek;
     const dueDateBeforeLast =
-      (weeklySummaries && weeklySummaries[2] && weeklySummaries[2].dueDate) ||
+      // (weeklySummaries && weeklySummaries[2] && weeklySummaries[2].dueDate) ||
```
Comment (rauleffting) [inline src/components/WeeklySummary/WeeklySummary.jsx:94]: Please, @LeeSingChanYau, delete these comments. Thereafter, I'm going to approve your PR! Thank you!

File: src/components/WeeklySummary/WeeklySummary.jsx [hunk 3]
```diff
@@ -70,33 +77,40 @@ export class WeeklySummary extends Component {
...
       this.state.dueDateBeforeLast;
+    const dueDateThreeWeeksAgo =
+      // (weeklySummaries && weeklySummaries[3] && weeklySummaries[3].dueDate) ||
```
Comment (rauleffting) [inline src/components/WeeklySummary/WeeklySummary.jsx:97]: Please, @LeeSingChanYau, delete these comments. Thereafter, I'm going to approve your PR! Thank you!

**Review by rauleffting — CHANGES_REQUESTED** [review]

**Comment by RafaelCastilhoCastro** [comment]
Hi @LeeSingChanYau,
I tested the PR again. No more freezing screen after submiting a summary. I did submit summaries to Current Week, Last Week, Two Weeks ago and Three Weeks Ago with all account roles with no issues.

The "Total submitted" is not working though. My admin account shows a total of 6 summaries submited, but it doesn't add to it when submitting new ones. It keeps to 0 on all my other accounts.

image

Thanks.

**Comment by one-community** [comment]
Good catch Rafael! Please fix that Alan.

**Comment by Filipe-Oliv** [comment]
Hi @LeeSingChanYau 

Same question as Rafael. The "Total submitted: 0" is not working.

week 03 - 02-09-23 (#622)

File: src/components/Dashboard/Dashboard.jsx [hunk 4]
```diff
@@ -48,6 +49,7 @@ export const Dashboard = props => {
         toggleSubmitForm={toggle}
         role={props.auth.user.role}
         leaderData={leaderData}
+        setCount={setCount}
```
Comment (PrabhjotSembhi) [inline src/components/Dashboard/Dashboard.jsx:52]: I think here you want to use 
`  setCount={weeklySummariesCount}`

correct me if i am wrong

**Review by PrabhjotSembhi — COMMENTED** [review]
As others are saying  Total submitted count is not getting updated. 
I have added comments in the code with potential issues. pleas have a look

**Comment by NyahJ** [comment]
Hi @LeeSingChanYau,

Most of the issues mentioned above are fixed, the features you changed went well with different accounts.
About the "Total submitted: 0", I also reading your codes and have the same view with PrabhjotSembhi, could you pls try it when you get a chance? Thanks!

截屏2023-02-23 上午11 23 34

**Comment by dkamghem** [comment]
In my case, i only have three tabs. i am not sure if it is what it is intended to be.
Screen Shot 2023-02-28 at 1 41 17 PM (2)

**Comment by RafaelCastilhoCastro** [comment]
Hi @dkamghem,
The UI on your image seems strange on some elements. Can you please make sure you are on the right branchs and has them update/installed (`git pull`/`npm i`) and double check your results? Thank you.

**Comment by Lunaliar** [comment]
Hey @LeeSingChanYau, Seems to be working for me! Great job!
Screenshot from 2023-03-01 10-14-56

**Comment by Lunaliar** [comment]
Hey @LeeSingChanYau, I checked out your changes, the tabs are working great and I'm able to write and view summaries from those weeks, however, the total submitted is still at 0, I tried following the code and debugging to see why its only showing at 0 but I keep losing track of it.
Screenshot from 2023-03-01 10-39-35
Screenshot from 2023-03-01 10-39-50

**Comment by ayush0904** [comment]
Hey @LeeSingChanYau, Total Submitted is always showing as 0. It is not getting updated. Please see to it. Thank you

Screen Shot 2023-03-04 at 3 32 17 PM

**Comment by EvianTan** [comment]
Close it due to I raised #707

**Pull request closed without merging**